packages/react-data-grid/src/Cell.js

```diff
@@ -247,115 +223,119 @@ export default class Cell extends React.PureComponent {
     }
 
     return allEvents;
-  };
+  }
 
-  getEvents = () => {
-    const columnEvents = this.props.column ? { ...this.props.column.events } : undefined;
-    const onColumnEvent = this.props.cellMetaData ? this.props.cellMetaData.onColumnEvent : undefined;
+  getEvents() {
+    const { column, cellMetaData } = this.props;
+    const columnEvents = column ? { ...column.events } : undefined;
+    const onColumnEvent = cellMetaData ? cellMetaData.onColumnEvent : undefined;
     const gridEvents = {
-      onClick: this.onCellClick,
-      onMouseDown: this.onCellMouseDown,
-      onMouseEnter: this.onCellMouseEnter,
-      onDoubleClick: this.onCellDoubleClick,
-      onContextMenu: this.onCellContextMenu,
-      onDragOver: this.onDragOver
+      onClick: this.handleCellClick,
+      onMouseDown: this.handleCellMouseDown,
+      onMouseEnter: this.handleCellMouseEnter,
+      onDoubleClick: this.handleCellDoubleClick,
+      onContextMenu: this.handleCellContextMenu,
+      onDragOver: this.handleDragOver
     };
 
     if (!columnEvents || !onColumnEvent) {
       return gridEvents;
     }
 
     return this.createEventDTO(gridEvents, columnEvents, onColumnEvent);
-  };
+  }
 
   getCellActions() {
     const { cellMetaData, column, rowData } = this.props;
-    if (cellMetaData && cellMetaData.getCellActions) {
+    if (isFunction(cellMetaData.getCellActions)) {
       const cellActionButtons = cellMetaData.getCellActions(column, rowData);
-      if (cellActionButtons && cellActionButtons.length) {
+      if (cellActionButtons && cellActionButtons.length > 0) {
         return cellActionButtons.map((action, index) => {
           return <CellAction key={index} action={action} isFirst={index === 0} />;
         });
       }
-      return null;
     }
     return null;
   }
 
-  setCellRef = (node) => {
-    this.node = node;
-  };
-
+  renderCellContent() {
+    let cellContent;
+    const { value, column, height, tooltip, isScrolling, isExpanded, expandableOptions, cellMetaData } = this.props;
+    const Formatter = column.formatter;
+    const cellProps = {
+      value,
+      isScrolling,
+      isExpanded,
+      column,
+      row: this.getRowData(),
+      dependentValues: this.getFormatterDependencies()
+    };
 
-  renderCellContent = (props) => {
-    let CellContent;
-    const Formatter = this.getFormatter();
     if (isElement(Formatter)) {
-      CellContent = React.cloneElement(Formatter, { ...props, dependentValues: this.getFormatterDependencies(), row: this.getRowData() });
+      cellContent = React.cloneElement(Formatter, cellProps);
     } else if (isValidElementType(Formatter)) {
-      CellContent = <Formatter value={this.props.value} dependentValues={this.getFormatterDependencies()} isScrolling={this.props.isScrolling} row={this.getRowData()} />;
+      cellContent = <Formatter {...cellProps} />;
```

Comment:
You can do `React.createElement(column.formatter, cellProps)`, that way you don't need to re-assign `column.formatter` just do to valid jsx.
Bonus point for looking similar to the `React.cloneElement` line. 🙂

Reply:
I will leave it as it for now but I will revisit it when Cell is migrated to TS